app/lib/features/news/pages/add_news_page.dart

```diff
@@ -302,20 +274,37 @@ class AddNewsState extends ConsumerState<AddNewsPage> {
   }
 
   Widget slideTextPostUI(BuildContext context) {
-    final backgroundColor = ref.watch(
-      newsStateProvider.select((i) => i.currentNewsSlide?.backgroundColor),
-    );
+    //Get document data
+    final document = selectedNewsPost!.html != null
+        ? ActerDocumentHelpers.fromHtml(selectedNewsPost!.html!)
+        : ActerDocumentHelpers.fromMarkdown(selectedNewsPost!.text ?? '');
+    textEditorState = EditorState(document: document);
```

Comment:
so here is a thing I have seen you do resonable bit that I thought was a no-no: changing widget state within the `build` function (or its sub function). I _thought_ that a state change like that triggers a rebuild and therefore it should only be done outside of `buildContext` areas as we change the underlying state _while_ trying to render it ... that's why this is usually offloaded in some side function or better even direct callbacks rather than implicit changes triggered through a ref.watch-update cycle ....

Now, you have replaced that offloaded code by doing it implicitly during render time and that feels ... odd at least. Your thoughts?

Reply:
alternative approach in #1418 .

Reply:
> alternative approach in https://github.com/acterglobal/a3/pull/1418 .

Thanks @gnunicorn for this.